homeassistant/components/point/config_flow.py

```diff
@@ -1,193 +1,51 @@
-        )
-
-        self.hass.http.register_view(MinutAuthCallbackView())
-
-        return point_session.get_authorization_url
-
-    async def async_step_code(self, code=None):
-        """Received code for authentication."""
-        if self._async_current_entries():
-            return self.async_abort(reason="already_setup")
-
-        if code is None:
-            return self.async_abort(reason="no_code")
-
-        _LOGGER.debug(
-            "Should close all flows below %s",
-            self._async_in_progress(),
-        )
-        # Remove notification if no other discovery config entries in progress
-
-        return await self._async_create_session(code)
-
-    async def _async_create_session(self, code):
-        """Create point session and entries."""
-
-        flow = self.hass.data[DATA_FLOW_IMPL][DOMAIN]
-        client_id = flow[CONF_CLIENT_ID]
-        client_secret = flow[CONF_CLIENT_SECRET]
-        point_session = PointSession(
-            async_get_clientsession(self.hass),
-            client_id,
-            client_secret,
-        )
-        token = await point_session.get_access_token(code)
-        _LOGGER.debug("Got new token")
-        if not point_session.is_authorized:
-            _LOGGER.error("Authentication Error")
-            return self.async_abort(reason="auth_error")
-
-        _LOGGER.info("Successfully authenticated Point")
-        user_email = (await point_session.user()).get("email") or ""
-
-        return self.async_create_entry(
-            title=user_email,
-            data={
-                "token": token,
-                "refresh_args": {
-                    CONF_CLIENT_ID: client_id,
-                    CONF_CLIENT_SECRET: client_secret,
-                },
-            },
-        )
-
-
-class MinutAuthCallbackView(HomeAssistantView):
-    """Minut Authorization Callback View."""
-
-    requires_auth = False
-    url = AUTH_CALLBACK_PATH
-    name = AUTH_CALLBACK_NAME
-
-    @staticmethod
-    async def get(request):
-        """Receive authorization code."""
-        hass = request.app[KEY_HASS]
-        if "code" in request.query:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN, context={"source": "code"}, data=request.query["code"]
-                )
-            )
-        return "OK!"
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(step_id="reauth_confirm")
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> ConfigFlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+        if existing_entry := await self.async_set_unique_id(DOMAIN):
+            self.hass.config_entries.async_update_entry(existing_entry, data=data)
+            await self.hass.config_entries.async_reload(existing_entry.entry_id)
```

Comment:
Should we make the reauth and the single entry setup a bit better? Check withings for an example. With the current PR its possible to create 2 entries and then this call will mess up big time